master/buildbot/worker/kubernetes.py

```diff
@@ -0,0 +1,240 @@
+                    kubeConfig=None, **kwargs):
+
+        # Set build_wait_timeout to 0 if not explicitly set: Starting a
+        # container is almost immediate, we can afford doing so for each build.
+        if 'build_wait_timeout' not in kwargs:
+            kwargs['build_wait_timeout'] = 0
+        if not client:
+            self.dependency_error()
+        self.load_config(kubeConfig)
+        AbstractLatentWorker.checkConfig(self, name, password, **kwargs)
+
+    @defer.inlineCallbacks
+    def reconfigService(self, name, password, job=None, namespace=None,
+                        masterFQDN=None, getMasterMethod=None,
+                        kubeConfig=None, **kwargs):
+
+        # Set build_wait_timeout to 0 if not explicitly set: Starting a
+        # container is almost immediate, we can afford doing so for each build.
+        if 'build_wait_timeout' not in kwargs:
+            kwargs['build_wait_timeout'] = 0
+        if password is None:
+            password = self.getRandomPass()
+        self.getMasterMethod = getMasterMethod
+        if masterFQDN is None:
+            masterFQDN = self.get_master_qdn
+        if callable(masterFQDN):
+            masterFQDN = masterFQDN()
+        self.masterFQDN = masterFQDN
+        self.load_config(kubeConfig)
+        self.kubeConfig = kubeConfig
+        self.namespace = namespace or 'default'
+        self.job = job or self.default_job()
+        masterName = unicode2bytes(self.master.name)
+        self.masterhash = hashlib.sha1(masterName).hexdigest()[:6]
+        yield AbstractLatentWorker.reconfigService(
+            self, name, password, **kwargs)
+
+    @defer.inlineCallbacks
+    def start_instance(self, build):
+        if self.instance is not None:
+            raise ValueError('instance active')
+        masterFQDN = self.masterFQDN
+        masterPort = '9989'
+        if self.registration is not None:
+            masterPort = str(self.registration.getPBPort())
+        if ":" in masterFQDN:
+            masterFQDN, masterPort = masterFQDN.split(':')
+        master_properties = Properties.fromDict({
+            'masterHash': (self.masterhash, self.properties_source),
+            'masterFQDN': (masterFQDN, self.properties_source),
+            'masterPort': (masterPort, self.properties_source),
+            'workerName': (self.name, self.properties_source),
+            'workerPass': (self.password, self.properties_source)
+        })
+        props = IProperties(build)
+        props.updateFromProperties(master_properties)
+        namespace = yield build.render(self.namespace)
+        job = yield build.render(self.job)
+        res = yield threads.deferToThread(
+            self._thd_start_instance,
+            namespace,
+            job
+        )
+        defer.returnValue(res)
+
+    def _thd_start_instance(self, namespace, job):
+        self.load_config(self.kubeConfig)
+        batch_client = client.BatchV1Api()
```

Comment:
please use httpclientservice instead of requiring install of kubernetes abstraction.

http://docs.buildbot.net/latest/developer/utils.html#buildbot-util-httpclientservice

see worker/marathon.py for example implementation and tests, it will not need threads anymore